src/devices/src/virtio/net/tap.rs

```diff
@@ -91,7 +99,7 @@ impl IfReqBuilder {
         // SAFETY: ioctl is safe. Called with a valid socket fd, and we check the return.
         let ret = unsafe { ioctl_with_mut_ref(socket, ioctl, &mut self.0) };
         if ret < 0 {
-            return Err(Error::IoctlError(IoError::last_os_error()));
+            return Err(Error::IfreqExecuteError(IoError::last_os_error()));
         }
```

Comment:
The return value should ideally be checked before we could mistakenly access it.

This can be done using a scope like:
```rust
        // SAFETY: ioctl is safe. Called with a valid socket fd, and we check the return.
        let ret = {
            let x = unsafe { ioctl_with_mut_ref(socket, ioctl, &mut self.0) };
            if x < 0 {
                return Err(Error::IfreqExecuteError(IoError::last_os_error()));
            }
            x
        };
```
or more simply, a match:
```rust
        // SAFETY: ioctl is safe. Called with a valid socket fd, and we check the return.
        let ret = match unsafe { ioctl_with_mut_ref(socket, ioctl, &mut self.0) } {
            x @ 0.. => x,
            _ => return Err(Error::IfreqExecuteError(IoError::last_os_error()));
        };
```

Reply:
We don't use `ret` anyway so according to your suggestion it should be 
```rust
        // SAFETY: ioctl is safe. Called with a valid socket fd, and we check the return.
        let _ = match unsafe { ioctl_with_mut_ref(socket, ioctl, &mut self.0) } {
            x @ 0.. => x,
            _ => return Err(Error::IfreqExecuteError(IoError::last_os_error()));
        };
```
but this just adds boilerplate code.

I think current check looks fine.

Reply:
Ok, 1st variant looks much better, keeps `unsafe` as small as possible.